### PR TITLE
Checks if user has sufficient neelams to sell in auctions

### LINKED ISSUE #187: Validate if you can create new auction
https://crypto.realdevsquad.com/auctions/new should check if you have enough objects you are auctioning

### PR DESCRIPTION
**Pull request by swarajpure**
Fixes #187

### CODE REVIEW

File: components/Auction/createNewAuction.js [hunk 1]
```diff
@@ -6,39 +6,70 @@ import fetchSelfDetails from '../../utils/fetchSelfDetails';
...
 
-  const [itemType, setItemType] = useState();
+  const [itemType, setItemType] = useState('neelam');
   const [quantity, setQuantity] = useState();
   const [initialPrice, setInitialPrice] = useState();
   const [duration, setDuration] = useState();
   const [isSubmitted, setIsSubmitted] = useState(false);
+  const [userMoney, setUserMoney] = useState();
 
   useEffect(() => {
```
Comment (SagarBajpai) [inline components/Auction/createNewAuction.js:21]: If both useEffect do not have any dependencies, then you can consider having one useEffect.  
  fetchSelfDetails() &   getUserWallet() can stay in one useEffect.

File: components/Auction/createNewAuction.js [hunk 2]
```diff
@@ -6,39 +6,70 @@ import fetchSelfDetails from '../../utils/fetchSelfDetails';
 
 const BASE_API_URL = process.env.NEXT_PUBLIC_BASE_API_URL;
 const AUCTIONS_URL = `${BASE_API_URL}/auctions`;
+const WALLET_URL = `${BASE_API_URL}/wallet`;
 
 const CreateNewAuction = () => {
   const router = useRouter();
 
-  const [itemType, setItemType] = useState();
+  const [itemType, setItemType] = useState('neelam');
```
Comment (SagarBajpai) [inline components/Auction/createNewAuction.js:14]: Using string directly can create typo error in future. Also, if there's a change in 'neelam' to some other coin, we'll end up replacing it at multiple places.
You can create a const coin = 'neelam' and use them everywhere.

File: components/Auction/createNewAuction.js (hunk 1 above)
Comment (swarajpure) [reply components/Auction/createNewAuction.js:23]: Done

File: components/Auction/createNewAuction.js (hunk 2 above)
Comment (swarajpure) [reply components/Auction/createNewAuction.js]: Done

**Review by SagarBajpai — APPROVED** [review]

**Review by harshith-venkatesh — DISMISSED** [review]

**Review by ankushdharkar — APPROVED** [review]

**Pull request merged**